### PR TITLE
SplunkPy - validate connection in test module in any case

### PR DESCRIPTION
**Pull request by ilappe**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/43505

## Description
Validate connection to Splunk in test module in any flow

### CODE REVIEW

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py [hunk 1]
```diff
@@ -1928,6 +1928,14 @@ def splunk_parse_raw_command():
 
 
 def test_module(service):
+
+    try:
+        # validate connection
+        service.info()
+    except AuthenticationError:
+        return_error('Authentication error, please validate your credentials')
```
Comment (anara123) [inline Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py:1936]: add unit test.

**Comment by esharf** [comment]
@ilappe please fix `SplunkPyPreRelease` integration also

**Review by yaakovi — APPROVED** [review]

File: Packs/SplunkPyPreRelease/ReleaseNotes/1_0_2.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### SplunkPy Prerelease (Beta)
+- Improved the test-module command.
```
Comment (yaakovi) [inline Packs/SplunkPyPreRelease/ReleaseNotes/1_0_2.md:4]:
```suggestion
- Improved the ***test-module*** command.
```

File: Packs/SplunkPy/ReleaseNotes/2_2_3.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### SplunkPy
+- Improved the test-module command.
```
Comment (yaakovi) [inline Packs/SplunkPy/ReleaseNotes/2_2_3.md:4]:
```suggestion
- Improved the ***test-module*** command.
```

**Review by yaakovi — APPROVED** [review]

**Pull request merged**